images/health-monitor/monitor.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import time
+
+import cloud_tpu_client
+
+from absl import app
+from absl import flags
+from absl import logging
+from kubernetes import client, config
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('pod', None, 'The name of the pod to monitor.')
+flags.DEFINE_string('namespace', 'default', 'The namespace of the pod.')
+flags.DEFINE_string('container', 'train', 'The name of the container to watch.')
+
+flags.DEFINE_integer('interval', 60, 'Number of seconds to wait between health checks.')
+flags.DEFINE_bool('verbose', False, 'Whether to print when TPU is HEALTHY')
+
+flags.DEFINE_string('project', None, 'The GCP project with your GKE cluster.')
+flags.DEFINE_string('zone', None, 'The GCP zone with your GKE cluster.')
+
+def main(_):
+  if FLAGS.verbose:
+    logging.set_verbosity(logging.DEBUG)
+  else:
+    logging.set_verbosity(logging.WARNING)
```

Comment:
does this mean that `logging.info` messages would not appear in Stackdriver?

If so, just make sure you're OK with this log message below being at `info` level:

```
      logging.info('Container `%s` terminated with status:\n%s',
                   FLAGS.container, str(status))
```

Reply:
That's a good point. I raised the log level because `cloud_tpu_client` produces quite a bit of log spam at the INFO level, and we didn't necessarily need to see that the TPU was healthy repeatedly. I think this last message could be useful, though. I'll go ahead and raise the log level of this to `warning`.